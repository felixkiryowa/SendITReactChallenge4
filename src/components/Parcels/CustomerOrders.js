import React from 'react'

const CustomerOrders = ({ parcelOrders }) => {
    return (
        <div className="container table-responsive{-sm|-md|-lg|-xl}" style={{ marginTop: "150px" }}>
            <table className="table">
                <thead className="thead-dark">
                    <tr>
                        <th scope="col">Pick up Address</th>
                        <th scope="col">Destination Address</th>
                        <th scope="col">Order Status</th>
                        <th scope="col">Price</th>
                        <th scope="col">Created At</th>
                        <th scope="col">Manage Order</th>
                    </tr>
                </thead>
                <tbody>
                    {
                        parcelOrders.length > 0 ? 
                            parcelOrders.map(order => {
                                return (
                                    <tr key={order.parcel_order_id}>
                                        <td>{order.parcel_pickup_address}</td>
                                        <td>{order.parcel_destination_address}</td>
                                        <td>{order.order_status}</td>
                                        <td>{order.price}</td>
                                        <td>{order.created_at}</td>
                                        <td>
                                            <div className="dropdown">
                                                <button
                                                    className="btn btn-success btn-sm dropdown-toggle"
                                                    type="button"
                                                    id="dropdownMenuButton"
                                                    data-toggle="dropdown"
                                                    aria-haspopup="true"
                                                    aria-expanded="false">
                                                    Action
                                                </button>
                                                <div className="dropdown-menu" aria-labelledby="dropdownMenuButton">
                                                    <a className="dropdown-item" href="#">Edit <i className="far fa-trash-alt"></i></a>
                                                    <a className="dropdown-item" href="#">Cancel <i className="far fa-window-close"></i></a>
                                                </div>
                                            </div>
                                        </td>
                                    </tr>
                                )
                            })
                            :
                            (
                                <tr>
                                    <td>
                                        <p align="center">
                                            <i className="no-orders">No Parcel Orders were found, Please create Order!!!!</i>
                                        </p>
                                    </td>
                                </tr>
                            )
                    }

                </tbody>
            </table>
        </div>
    )
}

export default CustomerOrders;
