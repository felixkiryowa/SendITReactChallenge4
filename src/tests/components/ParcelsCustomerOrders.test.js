import React from 'react';
import renderer from 'react-test-renderer';
import CustomerOrders from '../../components/Parcels/CustomerOrders';;

describe('Sample Articles Component', () => {
    let mountApp;
    const props = {
        parcelOrders: [
            {
                "created_at": "Tue, 28 May 2019 08:04:50 GMT",
                "order_current_location": "Kasese, Uganda",
                "order_name": null,
                "order_status": "pending",
                "parcel_destination_address": "Kampala, Uganda",
                "parcel_order_id": 1,
                "parcel_pickup_address": "Kasese, Uganda",
                "parcel_weight": 4,
                "price": 200000,
                "receivers_contact": null,
                "receivers_names": null,
                "senders firstname": "Roy ",
                "senders lastname": "Wasibani",
                "senders phone contact": "0700162509",
                "senders_user_id": 5
            },
            {
                "created_at": "Tue, 28 May 2019 08:04:50 GMT",
                "order_current_location": "Lugazi, Mukono, Uganda",
                "order_name": null,
                "order_status": "pending",
                "parcel_destination_address": "Jinja, Uganda",
                "parcel_order_id": 2,
                "parcel_pickup_address": "Lugazi, Mukono, Uganda",
                "parcel_weight": 5,
                "price": 250000,
                "receivers_contact": null,
                "receivers_names": null,
                "senders firstname": "Roy ",
                "senders lastname": "Wasibani",
                "senders phone contact": "0700162509",
                "senders_user_id": 5
            }
        ],
    };
    beforeEach(() => {
        mountApp = renderer.create(<CustomerOrders {...props} />);
    });

    it('Should render without crashing', () => {
        expect(mountApp).toMatchSnapshot();
    });
});
