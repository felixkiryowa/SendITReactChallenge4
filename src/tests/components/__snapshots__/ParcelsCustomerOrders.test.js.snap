// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample Articles Component Should render without crashing 1`] = `
<div
  className="container table-responsive{-sm|-md|-lg|-xl}"
  style={
    Object {
      "marginTop": "150px",
    }
  }
>
  <table
    className="table"
  >
    <thead
      className="thead-dark"
    >
      <tr>
        <th
          scope="col"
        >
          Pick up Address
        </th>
        <th
          scope="col"
        >
          Destination Address
        </th>
        <th
          scope="col"
        >
          Order Status
        </th>
        <th
          scope="col"
        >
          Price
        </th>
        <th
          scope="col"
        >
          Created At
        </th>
        <th
          scope="col"
        >
          Manage Order
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Kasese, Uganda
        </td>
        <td>
          Kampala, Uganda
        </td>
        <td>
          pending
        </td>
        <td>
          200000
        </td>
        <td>
          Tue, 28 May 2019 08:04:50 GMT
        </td>
        <td>
          <div
            className="dropdown"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              className="btn btn-success btn-sm dropdown-toggle"
              data-toggle="dropdown"
              id="dropdownMenuButton"
              type="button"
            >
              Action
            </button>
            <div
              aria-labelledby="dropdownMenuButton"
              className="dropdown-menu"
            >
              <a
                className="dropdown-item"
                href="#"
              >
                Edit 
                <i
                  className="far fa-trash-alt"
                />
              </a>
              <a
                className="dropdown-item"
                href="#"
              >
                Cancel 
                <i
                  className="far fa-window-close"
                />
              </a>
            </div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          Lugazi, Mukono, Uganda
        </td>
        <td>
          Jinja, Uganda
        </td>
        <td>
          pending
        </td>
        <td>
          250000
        </td>
        <td>
          Tue, 28 May 2019 08:04:50 GMT
        </td>
        <td>
          <div
            className="dropdown"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              className="btn btn-success btn-sm dropdown-toggle"
              data-toggle="dropdown"
              id="dropdownMenuButton"
              type="button"
            >
              Action
            </button>
            <div
              aria-labelledby="dropdownMenuButton"
              className="dropdown-menu"
            >
              <a
                className="dropdown-item"
                href="#"
              >
                Edit 
                <i
                  className="far fa-trash-alt"
                />
              </a>
              <a
                className="dropdown-item"
                href="#"
              >
                Cancel 
                <i
                  className="far fa-window-close"
                />
              </a>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
